ols/src/cache/postgres_cache.py

```diff
@@ -0,0 +1,142 @@
+"""Cache that uses Postgres to store cached values."""
+
+import pickle
+from typing import Optional
+
+from langchain_core.messages.base import BaseMessage
+
+from ols.src.cache.cache import Cache
+
+
+class PostgresCache(Cache):
+    """Cache that uses Postgres to store cached values."""
+
+    CREATE_CACHE_TABLE = """
+        CREATE TABLE IF NOT EXISTS cache (
+            user_id         text NOT NULL,
+            conversation_id text NOT NULL,
+            key             text UNIQUE NOT NULL,
+            value           bytea,
```

Comment:
`text` is not a good idea?

Reply:
Ah I see we are storing `BaseMessage`, not text.

Reply:
yup, it's serialized object basically